.github/workflows/tests.yml

```diff
@@ -55,36 +49,70 @@ jobs:
         # Sets outputs.appPath to the created directory and outputs.appName
         run: node ./packages/create-bison-app/scripts/createApp bison-ci-app --acceptDefaults
 
+      - name: Upload bison app
+        uses: actions/upload-artifact@v2
+        if: matrix.os == 'ubuntu-latest'
+        with:
+          name: ${{ steps.create-app.outputs.appName }}
+          path: |
+            ${{ steps.create-app.outputs.appPath }}/
+            !${{ steps.create-app.outputs.appPath }}/node_modules
+          retention-days: 1
+
+  test:
+    needs: create-app
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports: ["5432:5432"]
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x]
+
+    steps:
+      - name: Download bison app
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ needs.create-app.outputs.appName }}
+
+      - name: Set up node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install bison app packages
+        run: yarn install
+
       - name: Build production bison app
-        working-directory: ${{ steps.create-app.outputs.appPath }}
         run: yarn build
 
       - name: Lint bison app
-        working-directory: ${{ steps.create-app.outputs.appPath }}
         run: yarn lint
 
       - name: Set up test database
         env:
-          DATABASE_URL: postgresql://postgres:postgres@localhost/${{ steps.create-app.outputs.appName }}_test
+          DATABASE_URL: postgresql://postgres:postgres@localhost/${{ needs.create-app.outputs.appName }}_test
```

Comment:
Love when I learn a new part of the capabilities by reviewing a PR. `needs` is something I hadn't seen before.